pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/mergestrategy/MergeStrategyFactory.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.minion.mergestrategy;
+
+import java.util.Map;
+import org.apache.pinot.core.common.MinionConstants;
+
+
+public class MergeStrategyFactory {
+  public enum MergeStrategyType {
+    /**
+     * Time based merge strategy
+     */
+    TIME
+  }
+
+  private MergeStrategyFactory() {
+  }
+
+  public static MergeStrategy getMergeStrategy(Map<String, String> taskConfigs) {
+    String mergeStrategyTypeStr =
+        taskConfigs.getOrDefault(MinionConstants.MergeRollupTask.MERGE_STRATEGY_TYPE, MergeStrategyType.TIME.name());
+
+    MergeStrategyType mergeStrategyType;
+    try {
+      mergeStrategyType = MergeStrategyType.valueOf(mergeStrategyTypeStr.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      mergeStrategyType = MergeStrategyType.TIME;
```

Comment:
Should we abort for invalid value?

Reply:
I added the log and rethrow the exception